### PR TITLE
Feature/db data

### PR DESCRIPTION
**Pull request by l1s4un**
How to use:

1. configure config.js;
2. build project (npm run start);
3. run script to create  db (npm run dbcreate);
4. run script to insert some test data (npm run dbseed);

**If some table don't create, or you have error run script one more time.**

Now you have this tables in your local db:
screenshot from 2017-08-29 01-01-20

### CODE REVIEW

**Review by iuriifaevskii — APPROVED** [review]

File: helpers/pginit.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import pg from 'pg';
+import config from '../config.js';
+
+module.exports=function(){
+
+const conString = `postgres://${config.userName}:${config.password}@${config.dbServerLocation}:${config.serverPort}/${config.dbName}`;
```
Comment (python4eg) [inline helpers/pginit.js]: db creation and initialization should be separate tasks

File: helpers/pginit.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+const client = new pg.Client(conString);
+
+client.connect();
+client.query(`CREATE TABLE IF NOT EXISTS users(
```
Comment (python4eg) [inline helpers/pginit.js]: please read about query `on` and `end` methods

Comment (python4eg) [reply helpers/pginit.js]: error handling

File: helpers/pginit.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    tag_id          integer     references tags(id)
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS registartion(
+    id              varchar(36)   PRIMARY KEY,
+    email           varchar(50)   NOT NULL,
+    phone           varchar(24)   NOT NULL,
+    password        varchar(64)   NOT NULL,
+)`);
+
+client.query(`INSERT INTO users (fullname, userrole,passhesh, salt, email, is_premium, phone_number, is_deleted, avatar, aboutMe) VALUES
```
Comment (python4eg) [inline helpers/pginit.js]: all data should be stored as a separate object

File: helpers/pginit.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+client.query(`CREATE TABLE IF NOT EXISTS users(
+    id              serial        PRIMARY KEY,
+    fullname        varchar(64)   UNIQUE        NOT NULL,
+    userrole        varchar(20)   NOT NULL,
+    passhesh        varchar(40)   NOT NULL,
+    salt            varchar(16)   NOT NULL,
+    email           varchar(50)   NOT NULL,
+    is_premium      boolean       NOT NULL,
+    phone_number    varchar(24)   NOT NULL,
+    is_deleted      boolean       NOT NULL,
```
Comment (python4eg) [inline helpers/pginit.js]: nitpick: change with: `retire_date` to monitor when user's been retired/deleted

File: helpers/pginit.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    avatar          varchar(100),
```
Comment (python4eg) [inline helpers/pginit.js]: nice to have: change with `gravatar`

File: helpers/pginit.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+client.connect();
+client.query(`CREATE TABLE IF NOT EXISTS users(
+    id              serial        PRIMARY KEY,
+    fullname        varchar(64)   UNIQUE        NOT NULL,
+    userrole        varchar(20)   NOT NULL,
+    passhesh        varchar(40)   NOT NULL,
+    salt            varchar(16)   NOT NULL,
+    email           varchar(50)   NOT NULL,
+    is_premium      boolean       NOT NULL,
```
Comment (python4eg) [inline helpers/pginit.js]: change with type and create separate table with use types

File: helpers/pginit.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+module.exports=function(){
+
+const conString = `postgres://${config.userName}:${config.password}@${config.dbServerLocation}:${config.serverPort}/${config.dbName}`;
+const client = new pg.Client(conString);
+
+client.connect();
+client.query(`CREATE TABLE IF NOT EXISTS users(
+    id              serial        PRIMARY KEY,
+    fullname        varchar(64)   UNIQUE        NOT NULL,
+    userrole        varchar(20)   NOT NULL,
```
Comment (python4eg) [inline helpers/pginit.js]: should be `serial` and points to another table with all roles by id

File: helpers/pginit.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+client.query(`CREATE TABLE IF NOT EXISTS tags(
+    id           serial         PRIMARY KEY,
+    name         varchar(30)    UNIQUE      NOT NULL
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS resipe_tag(
+    id              serial      PRIMARY KEY,
+    resipe_id       integer     references ingradients(id),
+    tag_id          integer     references tags(id)
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS registartion(
```
Comment (python4eg) [inline helpers/pginit.js]: shy do we need this table?

File: helpers/pginit.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    id              varchar(36)   PRIMARY KEY,
+    email           varchar(50)   NOT NULL,
+    phone           varchar(24)   NOT NULL,
+    password        varchar(64)   NOT NULL,
+)`);
```
Comment (python4eg) [inline helpers/pginit.js]: please add postgresql like description to each field for all tables

File: helpers/pginit.js [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+client.query(`CREATE TABLE IF NOT EXISTS calc_card(
+    id              serial    PRIMARY KEY,
+    resipe_id       serial  references recipes(id),
+    ingredient_id   serial  references ingradients(id)
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS ingradient_map(
+    id              serial          PRIMARY KEY,
+    ingredient_id   integer         references ingradients(id), 
+    is_deleted      boolean         NOT NULL,
```
Comment (python4eg) [inline helpers/pginit.js]: add default value as FALSE

File: helpers/pginit.js [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    lat             real,
+    lon             real,
+    price           varchar(10)
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS tags(
+    id           serial         PRIMARY KEY,
+    name         varchar(30)    UNIQUE      NOT NULL
```
Comment (python4eg) [inline helpers/pginit.js]: tag description

File: helpers/pginit.js [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+)`);
+client.query(`CREATE TABLE IF NOT EXISTS resipe_tag(
```
Comment (python4eg) [inline helpers/pginit.js]: why do we need this extra layer?

**Review by python4eg — CHANGES_REQUESTED** [review]
please find all my comments above

File: helpers/pginit.js (hunk 1 above)
Comment (l1s4un) [reply helpers/pginit.js]: @python4eg Do you mean I should create a separate method for each task, or create a new script file for "initialization"?

File: helpers/pginit.js (hunk 8 above)
Comment (l1s4un) [reply helpers/pginit.js]: @python4eg One of the team members asks me to add this table for saving data about users who registered but don't confirm the email address. But now I think we can save this data in table `users` and set a default value of field `is_deleted` as TRUE and when user confirm his email address, then change this value. Isn't it?

File: helpers/pginit.js (hunk 11 above)
Comment (l1s4un) [reply helpers/pginit.js]: @python4eg What do you mean? Should I rename `name` like `tag description` or add a new field?

File: helpers/pginit.js (hunk 12 above)
Comment (l1s4un) [reply helpers/pginit.js]: @python4eg  We need this table for creating many-to-many relationship.

**Pull request merged**